File: Leetcode/66-Plus-One/PlusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        if (digits.size() == 0) {
            return digits;
        }
        if (digits[digits.size()-1]<9) {
            digits[digits.size()-1]++;
            return digits;
        }
        int i = digits.size()-1;
        while (i>=0) {
            if (digits[i]==9) {
                digits[i] = 0;
                i--;
            } else {
                digits[i]++;
                break;
            }
        }
        if (i==-1) {
            digits.insert(digits.begin(), 1);
        }
        return digits;
    }
};

File: Leetcode/66-Plus-One/PlusOne.py
if digits == None or len(digits)==0:
            return digits
        if digits[-1]<9:
            digits[-1]+=1
            return digits
        i = len(digits)-1
        while i>=0:
            if digits[i]==9:
                digits[i]=0
                i-=1
            else:
                digits[i]+=1
                break
        if i==-1:
            digits.insert(0, 1)
        return digits
        
